_search-plugins/sql/limitation.md

```diff
@@ -29,10 +29,24 @@ FROM (
 
 But, if the outer query has `GROUP BY` or `ORDER BY`, then it's not supported.
 
-## JOIN does not support aggregations on the joined result
+## JOIN queries
 
-The `join` query does not support aggregations on the joined result.
-For example, e.g. `SELECT depo.name, avg(empo.age) FROM empo JOIN depo WHERE empo.id == depo.id GROUP BY depo.name` is not supported.
+Because OpenSearch doesn't natively support relational operations, `JOIN` queries are supported on a best-effort basis.
+
+### JOIN does not support aggregations on the joined result
+
+The `JOIN` query does not support aggregations on the joined result.
+For example, `SELECT depo.name, avg(empo.age) FROM empo JOIN depo WHERE empo.id = depo.id GROUP BY depo.name` is not supported.
+
+### Performance
+
+`JOIN` queries are prone to expensive index scanning operations.
+Depending on the dataset, there may be scalability issues when running `JOIN` queries between data sets with more than around 5 million records.
```

Comment:
You mean JOIN query without any filter on two indices of 5m docs? Not sure what's changed but our previous performance test seems done with index larger than this?

Reply:
I ran the test without filters. I suspect without proof that we'd see the same results if we ran a join with filters, where the filter matches ~5mil records.

Reply:
I updated the doc to cover the hint and specify that it's based on result sets. This should be fine if we have push-down for `WHERE` clauses with joins.

Reply:
I've updated the doc with both workarounds. This does seem like expected behavior to me, since OS isn't relational we wouldn't expect this to perform well in the general case. In https://github.com/opensearch-project/sql/issues/3762 I discuss a few ways we could make this faster for some types of data.